core/src/avm2/object/class_object.rs

```diff
@@ -435,15 +434,20 @@ impl<'gc> ClassObject<'gc> {
     pub fn has_class_in_chain(self, test_class: ClassObject<'gc>) -> bool {
         let mut my_class = Some(self);
 
-        while let Some(class) = my_class {
-            if Object::ptr_eq(class, test_class) {
+        // A `ClassObject` stores all of the interfaces it implements,
+        // including those from superinterfaces and superclasses (recursively).
+        // Therefore, we only need to check interfaces once, and we can skip
+        // checking them when we processing superclasses in the `while`
+        // further down in this method.
```

Comment:
can we run this loop only if `test_class` is itself an interface?

Reply:
No - if `test_class` is a normal class, then this loop will check the implemented interfaces.

Reply:
I misread your comment - I'll add the check.